controllers/oidc.security/client_controller.go

```diff
@@ -496,6 +489,22 @@ func (r *ClientReconciler) processZenRegistration(ctx context.Context, req ctrl.
 		return subreconciler.RequeueWithError(err)
 	}
 
+	//Zen registration exist, update
+	if zenReg != nil {
+		//Zen registration exists - going to update zen instance registration
+		reqLogger.Info("Zen registration already exists for oidc client - the zen instance will be updated",
+			"clientId", clientCR.Spec.ClientId, "zenInstanceId", clientCR.Spec.ZenInstanceId)
+		if err = r.registerZenInstance(ctx, clientCR, clientCreds, config); err != nil {
+			reqLogger.Error(err, "Failed to update Zen registration")
+			if err := r.writeErrorConditionsAndEvents(ctx, clientCR, err, http.MethodPost); err != nil {
```

Comment:
should we use err2 here

Reply:
I think we should be good given for the initial error (err) we are calling the reconciler with error `return subreconciler.RequeueWithError(err)` , so the next check won't even happen. It is the same way being implemented in the existing flow.